xclim/sdba/properties.py

```diff
@@ -753,6 +753,68 @@ def _relative_frequency(
 )
 
 
+def _transition_probability(
+    da: xr.DataArray,
+    *,
+    initial_op: str = ">=",
+    final_op: str = ">=",
+    thresh: str = "1 mm d-1",
+    group: str | Grouper = "time",
+) -> xr.DataArray:
+    """Transition probability.
+
+    Probability of transition from the initial state to the final state. The states are
+     booleans comparing the value of the day to the threshold with the operator.
+
+     The transition occurs when consecutive days are both in the given states.
+
+    Parameters
+    ----------
+    da : xr.DataArray
+      Variable on which to calculate the diagnostic.
+    initial_op: {">", "<", ">=", "<="}
+      Operation to verify the condition for the initial state.
+      The condition is variable {op} threshold.
+    final_op: {">", "<", ">=", "<="}
+      Operation to verify the condition for the final state.
+      The condition is variable {op} threshold.
+    thresh: str
+      Threshold on which to evaluate the condition.
+    group : {'time', 'time.season', 'time.month'}
+      Grouping on the output.
+      Eg. For 'time.month', the relative frequency would be calculated on each month,
+      with all years included.
+
+    Returns
+    -------
+    xr.DataArray, [dimensionless]
+      Transition probability of values {initial_op} {thresh} to values {final_op} {thresh}.
+    """
+    # mask of the ocean with NaNs
+    mask = ~(da.isel({group.dim: 0}).isnull()).drop_vars(group.dim)
+
+    today = da.isel(time=slice(0, -1))
+    tomorrow = da.shift(time=-1).isel(time=slice(0, -1))
+
+    ops = {">": np.greater, "<": np.less, ">=": np.greater_equal, "<=": np.less_equal}
+    t = convert_units_to(thresh, da)
+    cond = ops[initial_op](today, t) * ops[final_op](tomorrow, t)
```

Comment:
For the comparing thing, I suggest using `xclim.indices.generic.compare`.

```suggestion
    # To be transported at the top
    from xclim.indices.generic import compare

    t = convert_units_to(thresh, da)
    cond = compare(today, initial_op, t) * compare(tomorrow, final_op, t)
```